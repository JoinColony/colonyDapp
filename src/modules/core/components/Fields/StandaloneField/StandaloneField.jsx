/* @flow */

import React, { Component } from 'react';
import type { ComponentType } from 'react';

type Props = {
  component: ComponentType<*>,
};

type State = {
  active: boolean,
};

class StandaloneField extends Component<Props, State> {
  setActive: () => void;
  setInactive: () => void;
  constructor(props: Props) {
    super(props);
    this.state = {
      active: false,
    };
    this.setActive = this.setActive.bind(this);
    this.setInactive = this.setInactive.bind(this);
  }
  setActive() {
    this.setState({
      active: true,
    });
  }
  setInactive() {
    this.setState({
      active: false,
    });
  }
  render() {
    const { component: RowComponent, ...props } = this.props;
    const { active } = this.state;
    return (
      <RowComponent {...props} onFocus={this.setActive} onBlur={this.setInactive} meta={{ active }} />
    );
  }
}

export default StandaloneField;
