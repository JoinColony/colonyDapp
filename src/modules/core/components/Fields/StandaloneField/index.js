/* @flow */

export { default } from './StandaloneField.jsx';
