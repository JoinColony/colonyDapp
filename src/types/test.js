/* @flow */

import {
  Address,
  ENSName,
  type AddressType,
  type ENSNameType,
} from './strings';

const takesENSName = (arg: ENSNameType) => arg;
const takesAddress = (arg: AddressType) => arg;

const name = ENSName('ens name');
const address = Address('0x123');

takesENSName(name);
takesAddress(address);

/*
 * The following should produce flow errors:
 */
// Address('not a hex string');
// takesAddress('0x123'); // just a string
// takesENSName('just a string');
