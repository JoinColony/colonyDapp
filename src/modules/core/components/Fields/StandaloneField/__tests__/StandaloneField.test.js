/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from '$utils/test';

import StandaloneField from '../StandaloneField.jsx';

const TestFieldComponent = () => <span />;

describe('StandaloneField component', () => {
  test('Renders initial component', () => {
    const wrapper = shallowWithIntl((
      <StandaloneField component={TestFieldComponent} help="halp" error="red" hasError={false} label="awesome" />
    ));
    expect(toJson(wrapper)).toMatchSnapshot();
    // component prop not present, but "intl" and "meta" added
    expect(Object.keys(wrapper.props())).toHaveLength(8);
  });
});
