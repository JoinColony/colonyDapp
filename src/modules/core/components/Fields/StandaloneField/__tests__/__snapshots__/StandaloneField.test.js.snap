// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandaloneField component Renders initial component 1`] = `
<TestFieldComponent
  error="red"
  hasError={false}
  help="halp"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "button.add": "Add",
        "button.back": "Back",
        "button.cancel": "Cancel",
        "button.confirm": "Confirm",
        "button.dismiss": "Dismiss",
        "button.next": "Next",
        "button.ok": "Okay",
        "button.remove": "Remove",
        "home": "Home",
        "input.resetText": "Cancel",
        "label.description": "Description",
        "label.email": "Email address",
        "label.name": "Name",
        "label.optional": "Optional",
        "label.password": "Password",
        "placeholder.email": "handle@domain.com",
        "placeholder.password": "Password",
        "role.admin": "Admin",
        "role.disabled": "Disabled",
        "role.member": "Member",
        "role.owner": "Owner",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  label="awesome"
  meta={
    Object {
      "active": false,
    }
  }
  onBlur={[Function]}
  onFocus={[Function]}
/>
`;
